src/ch09/Stack.java

```diff
@@ -0,0 +1,139 @@
+
+package ch09;
+
+class StackFullException extends Exception {
+	// not sure if the default value is better than suppressing the warning about it not being declared?
+	private static final long serialVersionUID = 1L;
+
```

Comment:
Might have been handy to have some sort of stringified version of the exception.